src/main/java/org/checkerframework/specimin/GetTypesFullNameVisitor.java

```diff
@@ -64,7 +64,15 @@ public Visitable visit(EnumDeclaration decl, Void p) {
   public Visitable visit(ClassOrInterfaceType type, Void p) {
     String typeFullName;
     try {
-      typeFullName = type.resolve().getQualifiedName();
+      /*
+       * In JavaParser, ClassOrInterfaceType is a subtype of ReferenceType
+       * (check an example here: https://github.com/javaparser/javaparser/blob/9c133d19d5b85b3b758f05762fb4d7c9875ef681/javaparser-core/src/main/java/com/github/javaparser/ast/type/ClassOrInterfaceType.java#L258).
+       * However, the resolve() method in ClassOrInterfaceType only returns a ResolvedType instead of a specific ResolvedReferenceType.
+       * This appears to be an inaccuracy within JavaParser's type hierarchy.
+       */
```

Comment:
I think this explanation should only appear once in the code. What I suggest you do is add a utility method of type `ClassOrInterfaceType -> ResolvedReferenceType` that has the comment, and then call at each of these locations.

Reply:
I have added a utility method. Please review again. Thank you.